### PR TITLE
Test setting the kamelet from an external file provided at runtime #3025

### PR DESCRIPTION
**Pull request by aldettinger**

### CODE REVIEW

File: integration-tests/kamelet/src/main/java/org/apache/camel/quarkus/component/kamelet/it/KameletRoutes.java [hunk 1]
```diff
@@ -62,6 +62,9 @@ public void configure() throws Exception {
         from("direct:chain")
                 .to("kamelet:echo/1?prefix=Camel Quarkus&suffix=Chained")
                 .to("kamelet:echo/2?prefix=Hello&suffix=Route");
+
+        from("direct:kamelet-location-at-runtime")
+                .kamelet("upper?location=file:src/test/resources/upper-kamelet.xml");
```
Comment (jamesnetherton) [inline integration-tests/kamelet/src/main/java/org/apache/camel/quarkus/component/kamelet/it/KameletRoutes.java:67]: Is that location going to work on `quarkus-platform`?

Comment (aldettinger) [reply integration-tests/kamelet/src/main/java/org/apache/camel/quarkus/component/kamelet/it/KameletRoutes.java:67]: Yeah good catch, it should not work on platform. Maybe it is feasible to find another way, worst case the test may be removed.

Comment (aldettinger) [reply integration-tests/kamelet/src/main/java/org/apache/camel/quarkus/component/kamelet/it/KameletRoutes.java]: The test is now using classpath, should be ok on quarkus-platform.

File: integration-tests/kamelet/pom.xml [hunk 2]
```diff
@@ -59,6 +59,10 @@
             <groupId>org.apache.camel.quarkus</groupId>
             <artifactId>camel-quarkus-bean</artifactId>
         </dependency>
+        <dependency>
+            <groupId>org.apache.camel.quarkus</groupId>
+            <artifactId>camel-quarkus-xml-io-dsl</artifactId>
```
Comment (jamesnetherton) [inline integration-tests/kamelet/pom.xml:64]: Can you add the `-deployment` dependency for this.

**Review by jamesnetherton — COMMENTED** [review]
Found another small thing to fix. Otherwise LGTM.

File: integration-tests/kamelet/pom.xml (hunk 2 above)
Comment (aldettinger) [reply integration-tests/kamelet/pom.xml:64]: Fixed, a `process-resources -P format -N` from root folder was missing.

**Pull request merged**